common/cache/lru.go

```diff
@@ -0,0 +1,97 @@
+package cache
+
+import "container/list"
+
+// NewLRUCache returns a new non-concurrent-safe LRU cache with input capacity
+func NewLRUCache(capacity uint64) *LRU {
+	return &LRU{
+		Cap:   capacity,
+		l:     list.New(),
+		items: make(map[interface{}]*list.Element),
+	}
+}
+
+// Add adds a value to the cache
+func (l *LRU) Add(key, value interface{}) {
+	if f, o := l.items[key]; o {
+		l.l.MoveToFront(f)
+		f.Value.(*item).value = value
+		return
+	}
+
+	newItem := &item{key, value}
+	itemList := l.l.PushFront(newItem)
+	l.items[key] = itemList
+	if l.Len() > l.Cap {
+		l.removeOldestEntry()
+	}
+}
+
+// Get returns keys value from cache if found
+func (l *LRU) Get(key interface{}) interface{} {
+	if i, f := l.items[key]; f {
+		l.l.MoveToFront(i)
+		return i.Value.(*item).value
+	}
+	return nil
+}
+
+// GetOldest returns the oldest entry
+func (l *LRU) GetOldest() (key, value interface{}) {
```

Comment:
Should GetOldest and Newest be exported? Not a nit, but I can't think of a use case that isn't for testing/internal use. Especailly since GetOldest doesn't bring it to the front which kinda circumvents LRU

Reply:
unexported both as they are not required and really just there for testing